diff --git a/snapcraft_legacy/internal/common.py b/snapcraft_legacy/internal/common.py
index 6017b405..d47d76cc 100644
--- a/snapcraft_legacy/internal/common.py
+++ b/snapcraft_legacy/internal/common.py
@@ -35,13 +35,14 @@ from snaphelpers import SnapConfigOptions, SnapCtlError
 from snapcraft_legacy.internal import errors
 
 SNAPCRAFT_FILES = ["parts", "stage", "prime"]
-_DEFAULT_PLUGINDIR = os.path.join(sys.prefix, "share", "snapcraft", "plugins")
+_SNAPCRAFT_LOCAL_DIR = os.path.join(Path.home(), ".local", "share", "snapcraft")
+_DEFAULT_PLUGINDIR = os.path.join(_SNAPCRAFT_LOCAL_DIR, "plugins")
 _plugindir = _DEFAULT_PLUGINDIR
-_DEFAULT_SCHEMADIR = os.path.join(sys.prefix, "share", "snapcraft", "schema")
+_DEFAULT_SCHEMADIR = os.path.join(_SNAPCRAFT_LOCAL_DIR, "schema")
 _schemadir = _DEFAULT_SCHEMADIR
-_DEFAULT_EXTENSIONSDIR = os.path.join(sys.prefix, "share", "snapcraft", "extensions")
+_DEFAULT_EXTENSIONSDIR = os.path.join(_SNAPCRAFT_LOCAL_DIR, "extensions")
 _extensionsdir = _DEFAULT_EXTENSIONSDIR
-_DEFAULT_KEYRINGSDIR = os.path.join(sys.prefix, "share", "snapcraft", "keyrings")
+_DEFAULT_KEYRINGSDIR = os.path.join(_SNAPCRAFT_LOCAL_DIR, "keyrings")
 _keyringsdir = _DEFAULT_KEYRINGSDIR
 
 _DOCKERENV_FILE = "/.dockerenv"
diff --git a/snapcraft_legacy/internal/dirs.py b/snapcraft_legacy/internal/dirs.py
index de3cb4d7..c5c16f06 100644
--- a/snapcraft_legacy/internal/dirs.py
+++ b/snapcraft_legacy/internal/dirs.py
@@ -14,7 +14,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import os.path
+import os
 import site
 import sys
 
@@ -120,4 +120,7 @@ def setup_dirs() -> None:
             common.get_keyringsdir(),
         ]:
             if not os.path.exists(d):
-                raise snapcraft_legacy.internal.errors.SnapcraftDataDirectoryMissingError()
+                try:
+                    os.makedirs(d, exist_ok=True)
+                except:
+                    raise snapcraft_legacy.internal.errors.SnapcraftDataDirectoryMissingError()
