diff --git a/craft_providers/bases/ubuntu.py b/craft_providers/bases/ubuntu.py
index 5af0fcc..e2d7cbc 100644
--- a/craft_providers/bases/ubuntu.py
+++ b/craft_providers/bases/ubuntu.py
@@ -20,6 +20,7 @@ import enum
 import io
 import logging
 import pathlib
+import platform
 import re
 import subprocess
 import sys
@@ -523,7 +524,7 @@ class BuilddBase(Base):
             )

             # don't inject snaps on non-linux hosts
-            if sys.platform != "linux" and not snap.channel:
+            if (sys.platform != "linux"  or "NixOS" in platform.version()) and not snap.channel:
                 raise BaseConfigurationError(
                     brief=(
                         f"cannot inject snap {snap.name!r} from host on "
