diff --git a/charmcraft/providers.py b/charmcraft/providers.py
index f6e7b1d..85466c0 100644
--- a/charmcraft/providers.py
+++ b/charmcraft/providers.py
@@ -19,6 +19,7 @@
 import enum
 import os
 import pathlib
+import platform
 import sys
 from typing import Dict, List, NamedTuple, Optional, Tuple, Union
 
@@ -180,8 +181,8 @@ def get_base_configuration(
     # injecting a snap on a non-linux system is not supported, so default to
     # install charmcraft from the store's stable channel
     snap_channel = get_managed_environment_snap_channel()
-    if snap_channel is None and sys.platform != "linux":
-        snap_channel = "stable"
+    if snap_channel is None and (sys.platform != "linux" or "NixOS" in platform.version()):
+        snap_channel = "candidate"
 
     base = get_base_from_alias(alias)
     charmcraft_snap = Snap(name="charmcraft", channel=snap_channel, classic=True)
