diff --git a/snapcraft/utils.py b/snapcraft/utils.py
index 77c76c6b..bc893976 100644
--- a/snapcraft/utils.py
+++ b/snapcraft/utils.py
@@ -87,28 +87,9 @@ def get_os_platform(
     ),
 ):
     """Determine a system/release combo for an OS using /etc/os-release if available."""
-    system = platform.system()
-    release = platform.release()
     machine = platform.machine()
-
-    if system == "Linux":
-        try:
-            with filepath.open("rt", encoding="utf-8") as release_file:
-                lines = release_file.readlines()
-        except FileNotFoundError:
-            emit.debug("Unable to locate 'os-release' file, using default values")
-        else:
-            os_release = {}
-            for line in lines:
-                line = line.strip()  # noqa PLW2901
-                if not line or line.startswith("#") or "=" not in line:
-                    continue
-                key, value = line.rstrip().split("=", 1)
-                if value[0] == value[-1] and value[0] in ('"', "'"):
-                    value = value[1:-1]
-                os_release[key] = value
-            system = os_release.get("ID", system)
-            release = os_release.get("VERSION_ID", release)
+    system = "ubuntu"
+    release = "22.04"
 
     return OSPlatform(system=system, release=release, machine=machine)
 
