diff --git a/tests/unit/linters/test_library_linter.py b/tests/unit/linters/test_library_linter.py
index 683439e0..8bcc4beb 100644
--- a/tests/unit/linters/test_library_linter.py
+++ b/tests/unit/linters/test_library_linter.py
@@ -33,7 +33,7 @@ def setup_function():
 
 def test_library_linter_missing_library(mocker, new_dir):
     """Verify missing libraries are caught by the linter."""
-    shutil.copy("/bin/true", "elf.bin")
+    shutil.copy(shutil.which("true"), "elf.bin")
 
     mocker.patch("snapcraft.linters.linters.LINTERS", {"library": LibraryLinter})
     mocker.patch(
@@ -129,7 +129,7 @@ def test_library_linter_unused_library(mocker, new_dir):
 )
 def test_library_linter_filter_missing_library(mocker, new_dir, filter_name):
     """Verify missing libraries can be filtered out."""
-    shutil.copy("/bin/true", "elf.bin")
+    shutil.copy(shutil.which("true"), "elf.bin")
 
     mocker.patch("snapcraft.linters.linters.LINTERS", {"library": LibraryLinter})
     mocker.patch(
